shtuff.py

```diff
@@ -21,15 +25,99 @@
     pass
 
 def main():
-    parser = argparse.ArgumentParser()
-    parser.add_argument('to_spawn', type=str, help='Command to spawn')
-    parser.add_argument('to_stuff', type=str, nargs='?', default='', help='Keys to stuff')
+    parser = argparse.ArgumentParser(
+        description="Shtuff will stuff commands into a shell à la tmux send-keys or screen stuff",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        epilog=dedent("""\
+            Examples:
+                In shell A, run:
+                  $ shtuff as shell-a
+
+                In shell B, run:
+                  $ shtuff into shell-a "git status"
+
+                Observe how shell A ran "git status"
+
+                An example use case for `shtuff new` might be a setup script to
+                open a couple shells automatically. Consider this script:
+                  #!/usr/bin/env bash
+                  termite -e "shtuff new vim" &
+                  termite -e "shtuff new 'tail -f /var/log/somelog.log'" &
+
+                This script will open two terminals, one running vim, and one
+                running tail.
+        """),
+    )
     parser.add_argument('-v', '--version', action='version', version=__version__)
-    args = parser.parse_args()
-    spawn_and_stuff(args.to_spawn, args.to_stuff)
+    subparsers = parser.add_subparsers(metavar="action")
+
+    parser_as = subparsers.add_parser('as', help='become a receiving shell')
+    parser_as.add_argument('name', help='the current shell will take this name, use `shtuff into` to send commands here')
+    parser_as.set_defaults(func=shtuff_as)
+
+    def add_newline_argument(parser):
+        parser.add_argument('-n', dest="newline", action='store_false', help="don't send a trailing newline, similar to echo")
+
+    parser_into = subparsers.add_parser('into', help='send a command to a receiving shell')
+    parser_into.add_argument('name', help='the name of the shell to send the given command to')
+    parser_into.add_argument('cmd', help='the command to send to the shell')
+    add_newline_argument(parser_into)
+    parser_into.set_defaults(func=shtuff_into)
+
+    parser_new = subparsers.add_parser('new', help='start a new shell and immediately send a command')
+    parser_new.add_argument('cmd', help='start a new shell and immediately run the given command')
+    add_newline_argument(parser_new)
+    parser_new.set_defaults(func=shtuff_new)
+
+
+    args = vars(parser.parse_args())
+    func = args.pop('func')
+    func(**args)
+
+def shtuff_as(name):
+    pid_file = get_pid_file(name)
+    pid = find_nearest_shtuff_process()
+
+    if not pid:
+        spawn_and_stuff(os.environ['SHELL'], f'shtuff as {name}\n')
+
+    with open(pid_file, 'w') as f:
+        f.write(str(pid))
+
+def shtuff_into(name, cmd, newline):
+    if newline:
+        cmd += "\n"
+
+    pid_file = get_pid_file(name)
+    with open(pid_file) as f:
+        pid = int(f.read().strip())
```

Comment:
this could be it's own method `def get_pid(name):`

Reply:
D'oh, this is a pretty obvious oversight!